lib/pbench/server/api/resources/query_apis/datasets/table_of_content.py

```diff
@@ -37,30 +57,27 @@ def assemble(self, json_data: JSON, context: CONTEXT) -> JSON:
 
         # Retrieve the ES indices that belong to this run_id from the metadata
         # table
-
         indices = self.get_index(dataset, "run-toc")
         if not indices:
             self.logger.debug(
                 f"Found no indices matching the prefix run-toc"
-                f"for a dataset {dataset!r}"
+                f" for a dataset {dataset!r}"
             )
             abort(HTTPStatus.NOT_FOUND, message="Found no matching indices")
 
         return {
             "path": f"/{indices}/_search",
             "kwargs": {
                 "json": {
-                    "size": 100,
                     "query": {
                         "bool": {
                             "filter": [
                                 {"match": {"run_data_parent": run_id}},
                                 {
                                     "dis_max": {
-                                        "queries": [
-                                            {"term": {"directory": parent}},
-                                            {"regexp": {"directory": f"{parent}[^/]+"}},
-                                        ]
+                                        "queries": {
+                                            "regexp": {"directory": f"{parent}[^/]+"}
+                                        }
                                     }
                                 },
                             ],
```

Comment:
`"regexp": {"directory": f"{parent}/[^/]+/[^/]+"}` using this which inclues a `/` after `{parent}` is listing all the directories under this directory.

Reply:
My approach for the situation is that the "target" directory (which I think we're calling `parent`) should be provided in the ES query _without_ a trailing slash (unless the caller is asking for the root of the tree, in which case `parent` would be `"/"`); the query should then select for two things (via `"dis_max"`) -- either the `"directory"` is equal to `parent`, or the directory is a path starting at `parent` -- and the query should also reject any documents where `"directory"` is a path pointing to a directory which is not a immediate subdirectory of `parent`.

Thus, the query should return the hit for the target directory (which contains the `files` list) as well as hits for the directories which have the target as their parent (but the query should not return any directories which have the target as the grand-parent or higher ancestor).

Then the job is the figure out which hit to get the `files` list from and to ignore it in all the other hits.

Reply:
By the way, in case anyone was wondering (i.e., in case I forget _again_), [ES/Lucene does not support anchors](https://www.elastic.co/guide/en/elasticsearch/reference/current/regexp-syntax.html#regexp-unsupported-operators) -- a regexp has to match the entire string.  (So...the placement of the trailing slash is quite important.)

Reply:
> Then the job is the figure out which hit to get the `files` list from and to ignore it in all the other hits.

Maybe this we can approach if we can assess the `parent` from assemle function. then we can check if `parent == directory` then we need to extract files else we need the name of sub-directory.